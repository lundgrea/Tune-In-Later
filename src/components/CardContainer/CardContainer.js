import React from 'react';
import Card from '../Card/Card'
import { connect } from 'react-redux';


const CardContainer = (props) => {
  let cards
  if(props.albums !== undefined) {
    cards = props.albums.map(album => {
    return <Card 
      artist={album.artistName}
      albumName={album.collectionName}
      genre={album.primaryGenreName}
      id={album.collectionId}
      img={album.collectionViewUrl}
      key={album.collectionId}
    />
  })
  } else {
    const response = "ENTER AN ARTIST"
    return response
  }


  return (
    <>
    {cards}
    </>
  )
}

const mapStateToProps = (store) => ({
  albums: store.albums
})

export default connect(mapStateToProps)(CardContainer)
