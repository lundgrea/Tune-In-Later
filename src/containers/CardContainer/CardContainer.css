.card-container {
  background: white;
  opacity: 0.8;
  min-height: 30vh;
  width: 30vh;
  flex: 1 0 25%;
  border: grey solid 2px;
  border-radius: 8px;
  margin: 15px;
  text-align: center;
}