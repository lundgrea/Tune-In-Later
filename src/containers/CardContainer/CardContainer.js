import React from 'react';
import Card from '../../containers/Card/Card'
import { connect } from 'react-redux';
import { toggleFavorite } from '../../actions'

const CardContainer = ({albums, toggleFavorite, user}) => {
  console.log(albums)
  let cards
  if(albums !== undefined) {
    cards = albums.map(album => {
    return <Card 
      {...album}
      toggleFavorite={toggleFavorite}
      user={user}
    />
  })
  } else {
    const response = "ENTER AN ARTIST"
    return response
  }


  return (
    <>
    {cards}
    </>
  )
}

const mapStateToProps = store => ({
  albums: store.albums,
  user: store.user
})


const mapDispatchToProps = dispatch => ({
  toggleFavorite: (id) => dispatch(toggleFavorite(id))
})


export default connect(mapStateToProps, mapDispatchToProps)(CardContainer)
