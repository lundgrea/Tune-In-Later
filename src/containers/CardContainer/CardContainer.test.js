import React from 'react';
// import  { shallow } from 'enzyme';
import CardContainer from './Cardcontainer';

describe('CardContainer', () => {
  it('should', () => {
   
//mapStateToProps

//mapDispatchToProps

//snapshot w/conditional rendering

  })
})

