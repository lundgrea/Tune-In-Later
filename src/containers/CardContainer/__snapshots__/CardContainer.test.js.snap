// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot 1`] = `
<Fragment>
  <div
    className="card-container"
  >
    <Connect(Card)
      album_id={12345}
      album_name="mirrorland"
      toggleFavorite={[MockFunction]}
      user={
        Object {
          "email": "aidanmckay2000@gmail.com",
          "id": 1,
          "name": "aidan",
        }
      }
    />
    <NavLink
      className="detail-button"
      key="12345"
      to="/12345"
    >
      Details
    </NavLink>
  </div>
  <div
    className="card-container"
  >
    <Connect(Card)
      album_id={123456}
      album_name="robots"
      toggleFavorite={[MockFunction]}
      user={
        Object {
          "email": "aidanmckay2000@gmail.com",
          "id": 1,
          "name": "aidan",
        }
      }
    />
    <NavLink
      className="detail-button"
      key="123456"
      to="/123456"
    >
      Details
    </NavLink>
  </div>
</Fragment>
`;
